05-html-css/flexbox-intro.md

```diff
@@ -0,0 +1,230 @@
+# Intro to Alignment and Positioning with Flexbox
+
+## Learning Goals
+- Understand the historical context of flexbox compared to other methods of aligning and positioning with CSS
+- Review how the browser uses _flow_ to place elements on the page
+- Change how elements flow on the page using CSS's `display: flex` property
+- Change how elements align on the page using CSS's `display: flex` property
+- Know browser compatibility for flexbox
+
+## Before We Begin: Review Display
+
+When rendering, the browser _flows_ from left to right, placing things in a single line. The line is as tall as the tallest element, and all the things are lined up with its bottom. If a thing would go off the edge of the screen, the browser wraps around and starts a new line, completely below the current one.
+
+By default, every HTML tag comes with a `display` property value of `inline` or `block`. The `display` property determines how elements flow on the page. As a general rule, flow has only be left to right, and top to bottom.
+
+Elements that are by `inline` elements align themselves left-to-right as many items that can fit into the container. Examples of these elements are `a`, `span`, and `img`.
+
+Elements that are `block` elements interrupt flow and start and end with a line break. Examples of these elements are `div`, `p`, and `h1`.
+
+Also, by default every element aligns the baseline of itself to the other baselines of surrounding elements.
+
+## History
+
+Historically, _aligning_ items while working with page flow using CSS has been difficult. CSS has never had a great way to align an item within along one dimension (vertically or horizontally). Vertical alignment in CSS is notoriously complex.
+
+In comes _flexbox_ (stylistically lowercase, sometimes "flex-box", short for Flexible Box Module)!
+
+Flexbox has the following advantages:
+1. More intuitive attributes for aligning compared to older CSS solutions (like vertical alignment or evenly spacing elements inside of a container)
+1. More relevant solutions to modern web layouts
+1. Built for responsive design
+1. Promotes thinking about alignment on websites agnostic of language direction
+
+If you choose to not use flexbox as your main way of aligning elements along one dimension, Ada has archived lectures on other ways to do layouts in CSS:
+1. [Layouts using Display](archived/layout-display.md)
+1. [Layouts using Floats](archived/layout-floats.md)
+1. [Layouts using Positioning](archived/layout-positioning.md)
+
+**Note:** Flexbox is fairly new, and adoption for using flexbox isn't widespread. A lot of information and advice on the Internet uses CSS that does not use flexbox, even if flexbox may be a better solution for that problem.
+
+## Rules for Setting Up Flexbox
+
+Flexbox solves the following problem: **How do I define rules for aligning many small elements in one larger container element?** Therefore, to use flexbox, we set up our flexbox with these rules:
+
```

Comment:
The last "flexbox" in this sentence should probably be a different word. As-is it's a little confusing.